Release notes for monitoring agent plugin 3.0.5 (#646)
128technology/docs: docs/plugin_monitoring_agent.md

## Monitoring Agent Plugin Notes

## Release 3.0.5

**Release Date:** Dec 14, 2023

**Router Version**

> - 128T-monitoring-agent-plugin-router-1.1.9-1

Reviewer: This is 1.1.10-2

- 128T-monitoring-agent-3.8.8-4

#### New Features and Improvements:

- **PLUGIN-630** Report additional LTE metrics

> The `t128_lte_collector` now includes additional information such as `carrier`, `connection-status`, `active-band-class`, `apn`, `service-mode` and `service-status`.

Reviewer 2: Does `t128_lte_collector` have any meaning to a customer? Would it be better to simply state:
Monitoring Agent now includes additional LTE information such as `carrier`, `connection-status`, `active-band-class`, `apn`, `service-mode` and `service-status`.

Author: Fair point @MichaelBaj - I'll re-word it!

- **PLUGIN-2275** Introduced `timeout` argument for `run-once` command

> A new `timeout` argument is added for the `run-once` testing tool provided by `monitoring-agent-cli` to control how long the input waits to complete its data completion.

Reviewer: completion -> collection?

#### Issues Fixed ####
  - **PLUGIN-2069 Disabling monitoring agent does not stop the collection services

    _**Resolution**_ The various monitoring agent services are now correctly stopped when the plugin is disabled or uninstalled.

>   - **PLUGIN-2274 Monitoring Agent Plugin allows `data-format` option on some inputs it shouldn't

Reviewer 3: ```suggestion
  - **PLUGIN-2274 Monitoring Agent Plugin incorrectly allows the `data-format` option on some output.
```
input and output do not need to be pluralized (is that a word??) They are deemed plural by context.

Reviewer 4: This isn't "output" in the sense of results from a script where the entity cannot be counted. In this case, the outputs are distinct, named, and countable. I believe `outputs` is correct.

    _**Resolution**_ For syslog output, the `data-format` option will no longer be allowed to avoid user confusion.

  - **I95-52139 High memory reported when using the cpu collector

    _**Resolution**_ The underlying memory leak for the CPU collector has been fixed to resolve the high memory condition.

>   - **PLUGIN-2272 The `include-output` configuration causes unrelated config to be added to the generated config

Reviewer: Unrelated > duplicated

>     _**Resolution**_ The code generation logic handles multiple `include-output` correctly and resolves the duplication in the generated config.

Reviewer 4: or instances/entries

  - **I95-53604 Router syslog output is sometimes malformed

>     _**Resolution**_ When an input uses multiple syslog output, the data corruption caused by message serialization code has been addressed.

Reviewer: Outputs

>     _**Resolution**_ When an input uses multiple syslog output, the data corruption caused by message serialization code has been addressed.

Author: > Isn't there a monitoring agent app section of release notes? Some of these should also go there?

I left the old notes there for posterity but I've stopped updating those since its a PITA.
